frontend/app/src/main/java/edu/pwr/iotmobile/androidimcs/ui/screens/main/MainScreen.kt

```diff
@@ -38,11 +50,20 @@ import org.koin.androidx.compose.koinViewModel
 fun MainScreen(navigation: MainScreenNavigation) {
     val viewModel: MainScreenViewModel = koinViewModel()
     val uiState by viewModel.uiState.collectAsState()
+    val context = LocalContext.current
 
     LaunchedEffect(navigation.isDashboardDeleted) {
         viewModel.updateLastAccessed()
     }
 
+    if (!ServiceManager.isServiceRunning(context) &&
```

Comment:
I don't think you should start this service here, this is just a UI screen, you should rather do it in MainActivity in onCreate